### PR TITLE
docs: add guidance on adding filtering and sorting logic to componetn…

### LINKED ISSUE #41652: no-filterpipe-or-orderbypipe appendix missing
# 📚 Docs or angular.io bug report

### Description

At some point between 2020/05/07 and 2020/09/23, the page https://angular.io/guide/pipes was updated to remove the no-filterpipe-or-orderbypipe appendix.  

While is true that in AngularJS (v1) there was a filter pipe and it was necessary to explain why there wasn't one in Angular v2, it's still possible to use pipes to filter arrays, and all the links to why this is "definitely discouraged" (Just one search on stackoverflow for "angular filter pipe" reveals a lot of them) are now pointing to a documentation that doesnt even mention filter pipes at all, which might lead to believe that this might be no longer discouraged.

It'd be a good idea to recover this section, even if it's no longer as important as it was back then, if only to avoid confusion on whether it still is discouraged

### PR DESCRIPTION
**Pull request by aikithoughts**
…s instead of pipes

Fixes #41652

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

File: aio/content/guide/styleguide.md [hunk 1]
```diff
@@ -2759,6 +2759,30 @@ A typical *lazy loaded folder* contains a *routing component*, its child compone
...
+
+<div class="s-rule avoid">
+
+**Avoid** adding filtering or sorting logic into custom pipes.
+
+</div>
+
+<div class="s-rule do">
+
+**Do** put filtering and sorting logic into components.
```
Comment (IgorMinar) [inline aio/content/guide/styleguide.md:2774]:
```suggestion
**Do** pre-compute the filtering and sorting logic in components or services before binding the model in templates.
```

Comment (IgorMinar) [reply aio/content/guide/styleguide.md:2774]: The key here is that the operations should be *precomputed*. It's not sufficient to just expose a method on the component that does sorting and filtering and then bind to that method from the template.

**Review by IgorMinar — APPROVED** [review]
One clarification, but otherwise LGTM! Thanks!

Reviewed-for: public-api

**Review by petebacondarwin — APPROVED** [review]
I concur with @IgorMinar 

Reviewed-for: global-docs-approvers

**Review by petebacondarwin — APPROVED** [review]
(Sorry I meant to approve for public-api, not global-docs-approvers)

Reviewed-for: public-api

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

reviewed-for: public-api

File: aio/content/guide/styleguide.md (hunk 1 above)
Comment (aikidave) [reply aio/content/guide/styleguide.md:2774]: Done!

**Pull request closed without merging**